test/e2e/generator/generate.go

```diff
@@ -134,8 +150,13 @@ func generateTestnet(r *rand.Rand, opt map[string]interface{}) (e2e.Manifest, er
 			startAt = nextStartAt
 			nextStartAt += 5
 		}
-		manifest.Nodes[fmt.Sprintf("full%02d", i)] = generateNode(
-			r, e2e.ModeFull, startAt, manifest.InitialHeight, false)
+		node := generateNode(r, e2e.ModeFull, startAt, manifest.InitialHeight, false)
+		if p2pNodeFactor == 0 {
+			node.UseNewP2P = manifest.UseNewP2P
+		} else if p2pNodeFactor%i == 0 {
+			node.UseNewP2P = !manifest.UseNewP2P
+		}
+		manifest.Nodes[fmt.Sprintf("full%02d", i)] = node
```

Comment:
I can't actually point to the correct piece of code but you've missed implementing this when we generate the validators